asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/JavaExtensionRegistryImpl.java

```diff
@@ -318,12 +328,31 @@ public JavaExtensionRegistry inlineMacro(String inlineMacroProcessor) {
         }
     }
 
-    private String getName(Class<?> clazz) {
-        Name nameAnnotation = clazz.getAnnotation(Name.class);
-        if (nameAnnotation == null) {
+    private List<Name> getNameAnnotations(List<Annotation> annotations, List<Annotation> visitedAnnotations) {
+        List<Name> allNameAnnotations = new ArrayList<>();
+        for (Annotation anno : annotations) {
+            if (anno.annotationType().isAssignableFrom(Name.class)) {
+                allNameAnnotations.add((Name) anno);
+            } else {
+                visitedAnnotations.add(anno);
+                List<Annotation> annotationsFromAnnotation = getAnnotationsFromAnnotationWithoutAlreadyVisitedAnnotations(anno, visitedAnnotations);
+                allNameAnnotations.addAll(getNameAnnotations(annotationsFromAnnotation, visitedAnnotations));
```

Comment:
Do we need the detection of duplicates?
If I understand it correctly it would only result in a relatively small optimisation in cases like this:
```java

@Target(TYPE)
@Retention(RUNTIME)
@Name("ABC")
public @interface MetaRoot {
    String NAME = "annotated";
}

@Target(TYPE)
@Retention(RUNTIME)
@MetaRoot
public @interface MetaOne {
    String NAME = "annotated";
}

@Target(TYPE)
@Retention(RUNTIME)
@MetaRoot
public @interface MetaTwo {
    String NAME = "annotated";
}

@MetaOne
@MetaTwo
class MyBlockProcessor extends AbstractBlockProcessor {...}
```

I think it is fine to fully iterate through the tree and collect the names into a set instead of a list.